system/include/emscripten/threading.h

```diff
@@ -296,9 +296,17 @@ int emscripten_dispatch_to_thread_async_(pthread_t target_thread,
 // Returns 1 if the current thread is the thread that hosts the Emscripten runtime.
 int emscripten_is_main_runtime_thread(void);
 
-// Returns 1 if the current thread is the main browser thread.
+// Returns 1 if the current thread is the main browser thread.  For historical
+// reasons this functions also returns 1 on the main node thread.
+// Use emscripten_thread_can_block() instead to determine if the current
+// thread can block.
 int emscripten_is_main_browser_thread(void);
 
+// Returns 1 if the current thread can use atomics.wait or other forms of
+// synchronous blocking.  This will return 0 on the main web browser thread
+// and 1 everywhere else, including on the main node thread.
+int emscripten_thread_can_block(void);
```

Comment:
Maybe a more precise name for this function could be `emscripten_thread_supports_atomics_wait()`? or `emscripten_thread_atomics_wait_allowed()`?
with a comment
```h
// Returns 1 if the calling thread can use the synchronously blocking atomics.wait() function.
```
That way if the APIs evolve, the function will retain a precise unambiguous meaning, and if a second function will need to be added for a hypothetical thing that works in browser but not in node or vice versa, this function will have a single exact semantic to associate with.

e.g. user might interpret a custom infinite `for(;;)` loop as one "forms of synchronous blocking", which the main node thread cannot support (or it can risk deadlocking other workers from proxied calls)

Reply:
Good point. 

Rather than changing anything in public API I will just make this new API internal for now.   I will call it `emscripten_thread_supports_atomics_wait` as you suggest but we don't need to commit to that since its internal-only for now.

I will update the PR title and description to match.